src/interface/lazy.jl

```diff
@@ -159,12 +160,12 @@ function Base.copyto!(dst::AbstractArray, src::LazyTensor{T, N}) where {T, N}
     return LazyTensor{T, N}(reformat(immediate(dst), src.data), src.extrude)
 end
 
-Base.permutedims(arg::LazyTensor{T, 2}) where {T} = permutedims(arg, [2, 1])
-function Base.permutedims(arg::LazyTensor{T, N}, perm) where {T, N}
+permutedims(arg::LazyTensor{T, 2}) where {T} = permutedims(arg, [2, 1])
+function permutedims(arg::LazyTensor{T, N}, perm) where {T, N}
     length(perm) == N || throw(ArgumentError("permutedims given wrong number of dimensions"))
     isperm(perm) || throw(ArgumentError("permutedims given invalid permutation"))
     perm = collect(perm)
-    idxs = [field(gensym(:i)) for _ in 1:N]
+    idxs = [field(Symbol(:i_, n)) for n = 1:N]
```

Comment:
this should stay as gensym, so as to avoid naming conflicts while we figure out scoping rules for the query language.

Reply:
Sure, done!